src/modules/joystick/sdl/Joystick.cpp

```diff
@@ -509,7 +515,7 @@ bool Joystick::setVibration(float left, float right, float duration)
 	Uint32 length = SDL_HAPTIC_INFINITY;
 	if (duration >= 0.0f)
 	{
-		float maxduration = std::numeric_limits<Uint32>::max() / 1000.0f;
+		float maxduration = std::numeric_limits<Uint32>::max() / 1000;
```

Comment:
An integer divided by an integer will implicitly do integer division, which isn't what we want here I think. To avoid the precision issues with converting uint32_max to a float, converting it to a double should work I believe:

```suggestion
		float maxduration = (float) (std::numeric_limits<Uint32>::max() / 1000.0);
```